daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/infoset/InfosetImpl.scala

```diff
@@ -1319,12 +1316,16 @@ sealed class DISimple(override val erd: ElementRuntimeData)
 
   def primType: NodeInfo.PrimType = erd.optPrimType.orNull
 
-  def contents: IndexedSeq[DINode] = IndexedSeq.empty
-
   private var _stringRep: String = null
   private var _bdRep: JBigDecimal = null
 
-  override def children: LazyList[DINode] = LazyList.empty
+  override def child(index: Int): DINode = null
```

Comment:
Does this want to be `Assert.invariant` or something? Nothing is probably going to check for null so it's just going to propogate elsewhere and lead to a bug. The other functions are probably fine since they return values that will usually be checked.